charmcraft/commands/pack.py

```diff
@@ -253,3 +329,81 @@ def _pack_bundle(self, parsed_args) -> List[pathlib.Path]:
             build.launch_shell()
 
         return [zipname]
+
+    def _recursive_pack_bundle(
+        self,
+        parsed_args: argparse.Namespace,
+        bundle_config: Optional[Dict[str, Any]],
+        charms: Mapping[str, pathlib.Path],
+    ) -> List[pathlib.Path]:
+        bundle_charms = bundle_config.get("applications", {})
```

Comment:
curious that this typechecked, can't `bundle_config` be `None` here?

Reply:
Charmcraft's checks don't currently include type checking, though I'm working towards that in #1062 . Currently though, `pyright` produces 262 errors and 1 warning, and it'll be fairly heavy lifting to get things corrected. 

I've corrected the typing issue (made `bundle_config` non-optional and ensured it's always a dict in `run`).